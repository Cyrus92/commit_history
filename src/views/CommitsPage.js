import React from 'react';
import ProjectSwitcher from '../component/ProjectSwitcher';
import CommitList from '../component/CommitList.js';

export default class CommitsPage extends React.Component {
  constructor(props) {
    super(props);
    // Set initial states
    this.state = {
      commitsData:""
    };    
    
    // bind  setCommitsData 
    this.setCommitsData = this.setCommitsData.bind(this);
    
}

componentDidMount() {
  
  
}

// set CommitsData  state
setCommitsData(data) {
  this.setState({commitsData: data})
}

  render() {
    return (
      <div>
          {/* ProjectSwitcher component */}
          <ProjectSwitcher  setCommitsData={this.setCommitsData} />
          
          {/* CommitList component */}
          <CommitList commitsData={this.state.commitsData}    />
      </div>
    )
  }
}