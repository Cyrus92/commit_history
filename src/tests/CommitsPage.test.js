import React from 'react';
import { shallow, mount } from 'enzyme';
import CommitsPage from '../views/CommitsPage';
describe("CommitsPage component", () => {

    it("should render", () => {
        const wrapper = shallow(<CommitsPage />);
    });

    it("should render initial layout", () => {
        // when
        const component = shallow(<CommitsPage />);
        // then
        expect(component.getElements()).toMatchSnapshot();
    });

    it("should update commitsData to match commits in ProjectSwitcher after calling setCommitsData in ProjectSwitcher" , () => {
        // given
        const component = mount(<CommitsPage />);
        const ProjectSwitcherComponent = component.find('ProjectSwitcher');
        const instance = ProjectSwitcherComponent.instance();
        // when
        ProjectSwitcherComponent.props().setCommitsData("data");
        // then
        expect(component.state('commitsData')).toEqual('data');
    });
    
    

});