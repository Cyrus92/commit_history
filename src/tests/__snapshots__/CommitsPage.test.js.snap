// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommitsPage component should render initial layout 1`] = `
Array [
  <div>
    <ProjectSwitcher
      setCommitsData={[Function]}
    />
    <CommitList
      commitsData=""
    />
  </div>,
]
`;
